src/cli/install.sh

```diff
@@ -169,129 +161,141 @@ fi
                 echo -e '\n# bun'
 
                 for command in "${commands[@]}"; do
                     echo "$command"
                 done
-            } >>"$bash_config"
+            } >>"$zsh_config"
 
-            info "Added \"$tilde_bin_dir\" to \$PATH in \"$tilde_bash_config\""
+            info "Added \"$tilde_bin_dir\" to \$PATH in \"$tilde_zsh_config\""
 
-            refresh_command="source $bash_config"
-            set_manually=false
-            break
+            refresh_command="exec $SHELL"
+        else
+            echo "Manually add the directory to $tilde_zsh_config (or similar):"
+
+            for command in "${commands[@]}"; do
+                info_bold "  $command"
+            done
         fi
-    done
+        ;;
+    bash)
+        # Install completions, but we don't care if it fails
+        IS_BUN_AUTO_UPDATE=true SHELL=bash $exe completions &>/dev/null || :
+
+        commands=(
+            "export $install_env=$quoted_install_dir"
+            "export PATH=$bin_env:\$PATH"
+        )
+
+        bash_configs=(
+            "$HOME/.bashrc"
+            "$HOME/.bash_profile"
+        )
+
+        if [[ ${XDG_CONFIG_HOME:-} ]]; then
+            bash_configs+=(
+                "$XDG_CONFIG_HOME/.bash_profile"
+                "$XDG_CONFIG_HOME/.bashrc"
+                "$XDG_CONFIG_HOME/bash_profile"
+                "$XDG_CONFIG_HOME/bashrc"
+            )
+        fi
+
+        set_manually=true
+        for bash_config in "${bash_configs[@]}"; do
+            tilde_bash_config=$(tildify "$bash_config")
+
+            if [[ -w $bash_config ]]; then
+                {
+                    echo -e '\n# bun'
+
+                    for command in "${commands[@]}"; do
+                        echo "$command"
+                    done
+                } >>"$bash_config"
 
-    if [[ $set_manually = true ]]; then
-        echo "Manually add the directory to $tilde_bash_config (or similar):"
+                info "Added \"$tilde_bin_dir\" to \$PATH in \"$tilde_bash_config\""
 
-        for command in "${commands[@]}"; do
-            info_bold "  $command"
+                refresh_command="source $bash_config"
+                set_manually=false
+                break
+            fi
         done
-    fi
-    ;;
-*)
-    echo 'Manually add the directory to ~/.bashrc (or similar):'
-    info_bold "  export $install_env=$quoted_install_dir"
-    info_bold "  export PATH=\"$bin_env:\$PATH\""
-    ;;
-esac
+
+        if [[ $set_manually = true ]]; then
+            echo "Manually add the directory to $tilde_bash_config (or similar):"
+
+            for command in "${commands[@]}"; do
+                info_bold "  $command"
+            done
+        fi
+        ;;
+    *)
+        echo 'Manually add the directory to ~/.bashrc (or similar):'
+        info_bold "  export $install_env=$quoted_install_dir"
+        info_bold "  export PATH=\"$bin_env:\$PATH\""
+        ;;
+    esac
+}
+
+if [ -z ${!install_env} ]; then
```

Comment:
this syntax is valid? shouldn't this be `if [ -z "${install_env}" ]`?

Reply:
This is an indirect variable reference, I'm checking if BUN_INSTALL is defined or not. `install_env` is always defined as it's defined in this script with a default.

I could use `BUN_INSTALL` directly, but it was only mentioned once, when reading the default value for `install_env`, and I didn't dare write a variables value twice.

Reply:
I didn't add double quotes as it's referring to an env variables name and that should be a continuous string, not expected to split